database/scrapers/departments.ts

```diff
@@ -0,0 +1,186 @@
+          `https://admin.topwr.solvro.pl/assets/${departmentEntry.logo}`,
+          `file ${departmentEntry.logo}`,
+        );
+
+        if (fileResponse.body === null) {
+          throw new Error(
+            `Response body is null for department ${departmentEntry.id} with asset id ${departmentEntry.logo}`,
+          );
+        }
+        const name = await this.filesService.uploadStream(
+          Readable.fromWeb(fileResponse.body),
+          extension,
+        );
+
+        // Address
+        const match = this.addressRegex.exec(departmentEntry.address);
+        if (match !== null) {
+          const postalCodeAndCity = match[0];
+          const addressLine1 = departmentEntry.address
+            .replace(this.addressRegex, "")
+            .trim();
+          const addressLine2 = postalCodeAndCity;
+          return {
+            id: departmentEntry.id,
+            name: departmentEntry.name,
+            addressLine1,
+            addressLine2,
+            code: departmentEntry.code,
+            betterCode: departmentEntry.betterCode,
+            logo: name,
+            description: departmentEntry.description,
+            gradientStart: departmentEntry.gradient_start,
+            gradientStop: departmentEntry.gradient_end,
+            createdAt: DateTime.now(),
+            updatedAt: DateTime.now(),
+          };
+        } else {
+          this.logger.warning(
+            `Skipped record ${departmentEntry.id} because ${departmentEntry.address} doesn't match regex.`,
+          );
+          return undefined;
+        }
+      }),
+    );
+
+    await DepartmentModel.createMany(
+      departmentEntries.filter((entry) => entry !== undefined),
+    );
+    task.update("Departments created!");
+
+    const formattedFieldsOfStudies = await Promise.all(
+      fieldOfStudyData.data.map(async (data) => {
+        return {
+          id: data.id,
+          departmentId: data.department_id ?? undefined,
+          name: data.name,
+          url: data.url,
+          isEnglish: data.isEnglish,
+          is2ndDegree: data.is2ndDegree,
+          semesterCount: data.isLongCycleStudies ? 12 : 7,
+          hasWeekendOption: data.hasWeekendModeOption,
+          createdAt: DateTime.now(),
+          updatedAt: DateTime.now(),
+        };
+      }),
+    );
+
+    await FieldOfStudyModel.createMany(formattedFieldsOfStudies);
+
+    task.update("Fields of Studies created!");
+    await DepartmentLinkModel.createMany(
+      departmentLinkData.data
+        .filter((linkEntry) => {
+          if (linkEntry.department_id === null) {
+            this.logger.warning(
+              `Skipped link entry ${linkEntry.id} due to missing department_id.`,
+            );
+            return false;
+          }
+          return true;
+        })
+        .map((linkEntry) => {
+          return {
+            id: linkEntry.id,
+            departmentId: linkEntry.department_id ?? undefined,
```

Comment:
`departmentId` cannot be undefined here, but it should be safe to assert that it is not `null` - this is a TS moment.